lib/correios/cep/web_service.rb

```diff
@@ -23,43 +24,12 @@ def initialize
       end
 
       def request(zipcode)
-        http = build_http
-
-        request = build_request(zipcode)
-        Correios::CEP.log_request(request, uri.to_s)
-
-        response = http.request(request)
-        Correios::CEP.log_response(response)
-
-        http.finish if http.started?
-
-        response.body
-      end
-
-      private
-
-      attr_reader :uri, :proxy_uri
-
-      def build_http
-        Net::HTTP.start(
-          uri.host,
-          uri.port,
-          proxy_uri.host,
-          proxy_uri.port,
-          nil,
-          nil,
-          use_ssl: true,
-          verify_mode: OpenSSL::SSL::VERIFY_NONE,
-          open_timeout: Correios::CEP.request_timeout,
-          read_timeout: Correios::CEP.request_timeout
-        )
-      end
-
-      def build_request(zipcode)
-        request = Net::HTTP::Post.new(uri.path)
-        request['Content-Type'] = CONTENT_TYPE_HEADER
-        request.body = BODY_TEMPLATE % { zipcode: zipcode }
-        request
+        HTTP
+          .use(logging: {logger: Logger.new(STDOUT)})
+          .headers('Content-Type' => CONTENT_TYPE_HEADER)
+          .post(@uri.to_s, body: BODY_TEMPLATE % { zipcode: zipcode })
```

Comment:
Please extract a method that generates the body.

Reply:
You mean all the code between row 27 and 30 or just the `.body.to_s`?

Reply:
I mean:
```ruby
BODY_TEMPLATE % { zipcode: zipcode }
```